### PR TITLE
chore: new fluid tokens

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
POC for new fluid tokens

### CODE REVIEW

File: app/client/packages/design-system/theming/src/theme/src/useTheme.tsx [hunk 1]
```diff
@@ -19,26 +19,30 @@ export function useTheme(props: UseThemeProps = {}) {
   const {
     borderRadius,
     colorMode = "light",
+    densityRatio = 1,
```
Comment (ichik) [inline app/client/packages/design-system/theming/src/theme/src/useTheme.tsx:22]: Would it make sense to name theme props just `density` and `sizing` to not be confused with the ratios used inside the scales?

**Review by ichik — APPROVED** [review]
Numbers need some fine-tuning (I'll get on on that), but overall great.

File: app/client/packages/design-system/theming/src/theme/src/useTheme.tsx (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/theme/src/useTheme.tsx:22]: If we name it just `sizing`, then we will have a name conflict with tokens which is returned from useFluidTokens.

```
const { innerSpacing, sizing, spacing, typography } = useFluidTokens(
    fluid,
    densityRatio,
    sizingRatio,
  );
```

`densityRatio` is named by analogy with `sizingRatio`

File: app/client/packages/design-system/theming/src/token/src/fluid/types.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 export interface FluidConfig {
   minVw: number;
   maxVw: number;
-  rootUnit: ScaleConfig;
+  sizing: ScaleConfig;
   spacing: ScaleConfig;
+  innerSpacing: ScaleConfig;
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/token/src/fluid/types.ts:6]: I was thinking about `outerSpacing`, but decided to leave the simple name. I think it's still worth renaming. @jsartisan  what do you think?

File: app/client/packages/design-system/theming/src/theme/src/useTheme.tsx (hunk 1 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/theme/src/useTheme.tsx:22]: Hm, maybe `themeSizing` or something like that then?

File: app/client/packages/design-system/theming/src/token/src/fluid/getFluidSpacing.ts [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
-  const { maxV, minV, ...rest } = spacing;
+  const {
+    maxV,
+    minV,
+    userDensityRatio = 1,
+    userSizingRatio = 1,
+    ...rest
+  } = spacing;
+
+  const ratio = userDensity * userDensityRatio + userSizing * userSizingRatio;
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/token/src/fluid/getFluidSpacing.ts:19]: @ichik Pay attention to these calculations.

File: app/client/packages/design-system/theming/src/token/src/defaultTokens.json [hunk 4]
```diff
@@ -3,35 +3,53 @@
   "fluid": {
     "minVw": 360,
     "maxVw": 1920,
-    "rootUnit": {
-      "minV": 4,
-      "maxV": 5,
+    "sizing": {
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/token/src/defaultTokens.json:6]: @ichik take a look at the config please.

File: app/client/packages/design-system/theming/src/token/src/fluid/getFluidSpacing.ts (hunk 3 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/token/src/fluid/getFluidSpacing.ts:19]: @KelvinOm Why addition and not multiplication of everything?

File: app/client/packages/design-system/theming/src/token/src/defaultTokens.json (hunk 4 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/token/src/defaultTokens.json:6]: `user*` kind of creates an expectation of it being adjustable by the end-user and not just by the builder. Maybe use `ui*`?

**Review by ichik — APPROVED** [review]

**Pull request merged**